File: Assignment5/Clause.java
class Clause {
  Clause(final int len, final int[] hgs, final int base, final int neck, final int[] xs) {
    this.hgs = hgs; // head+goals pointing to cells in cs
    this.base = base; // heap where this starts
    this.len = len; // length of heap slice
    this.neck = neck; // first after the end of the head
    this.xs = xs; // indexables in head
    
    
    System.out.println("Length:"+len);
    System.out.println("Base:"+base);
    System.out.println("Index XS:");
    for(int i=0; i<xs.length;i++)
    System.out.print("  "+xs[i]);
    System.out.println();
    System.out.println("Value of the neck:"+neck);
    System.out.println("Heap: ");
    for(int i=0; i<hgs.length;i++)
        System.out.print("  "+hgs[i]);
  }
  final int len; 
  final int[] hgs;
  final int base;
  final int neck;
  final int[] xs;
 
  }
public class ClauseTest{
	public static void main(String args[])
	{
		int [] ab={1,2,4};
		int [] bc={6,7,9};
		Clause cl=new Clause(10,ab, 7,3 ,bc);
		
	}

}

File: Assignment5/Clause.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;
class Clause 
{
public:
  int len;
  int * hgs;
  int base;
  int neck;
  int * xs;
  
  Clause( int len, int *hgs, int base, int neck,int *xs)
  {
    this->hgs=hgs; // head+goals pointing to cells in cs
    this->base=base; // heap where this starts
    this->len=len; // length of heap slice
    this->neck=neck; // first after the end of the head
    this->xs= xs; // indexables in head
    cout<<"Length :";
    cout<<len;
    cout<<"\n";
    cout<<"Base ";
    cout<<base;
    cout<<"\n";
    cout<<"Index xs : \n";
    for(int i=0;i<3;i++)
{
    cout << hgs[i];
    cout<<" ";
   
}
  cout<<"\n";
  cout<<"Neck : ";
  cout<<neck; 
   cout<<"\n";
    cout<<"Heap : \n";
    for(int j=0;j<3;j++)
{
    cout << xs[j];
    cout<<" ";
   
}
    
  }
};
int main()
{
int str1[]={1,2,4};
int str3[]={6,7,9};
int x;
Clause b(10,&str1[0],7,3,&str3[0]);
 
  cin>>x;
  return 0;
}

